doc/KUKSA.val_data_broker/authorization.md

```diff
@@ -0,0 +1,505 @@
+principal owner of a resource has certain access rights that needs to be verified by the
+resource server. This can be accomplished using the user id, group membership, roles or other entitlements present in the access token as claims.
+
+In addition to this, the resource owner can further limit what's authorized by an access
+token by specifying a scope in the request to the authorization server. The issued access
+token will then only grant this subset of authorization to the client (e.g. a third party app)
+using the token.
+
+There are currently no plans in KUKSA.VAL to separately verify the access rights of the
+resource owner itself, or what kind of access rights a certain group membership or role would provide. What this means in practice is that anyone authorized to get a valid access token is assumed to have full access rights within KUKSA.VAL, which will only be limited by the scope provided.
+
+There is nothing in the design preventing KUKSA.VAL from including support for other types
+of verifications in the future. The scope as specified here will continue to work as a way to limit the access granted by these other means anyway.
+
+
+### Limit access to resources
+The issued access token will typically be intended to for use with a certain resource. In the
+context of KUKSA.VAL, this can be a certain vehicle, or a certain backend service.
+
+To prevent access tokens issued for one resource from being used to gain access to other,
+unintended resources, OAuth2 introduces the concept of [Resource Indicators [RFC 8707]](https://datatracker.ietf.org/doc/html/rfc8707). By specifying the intended resource in the request
+to the authorization server, the resulting `"aud"` or audience claim should be populated with
+this resource.
+
+From [RFC9068](https://datatracker.ietf.org/doc/html/rfc9068#JWTATLRequest):
+> If the request includes a "resource" parameter (as defined in [[RFC8707](https://datatracker.ietf.org/doc/html/rfc6749)]), the resulting JWT access token "aud" claim SHOULD have the same value as the "resource" parameter in the request.
+
+and from [RFC8707](https://datatracker.ietf.org/doc/html/rfc6749):
+> To prevent cross-JWT confusion, authorization servers MUST use a distinct identifier as an "aud" claim value to uniquely identify access tokens issued by the same issuer for distinct resources. For more details on cross-JWT confusion, please refer to Section 2.8 of [RFC8725].
+
+By using a resource identifier that uniquely identifies the resource (i.e. a certain vehicle),
+and including this in the `"aud"` claim, KUKSA.VAL can verify that the access token isn't meant
+for another resource.
+
+KUKSA.VAL also needs to verify that the access token is intended to be used for KUKSA.VAL.
+An `"aud"` claim that would address both these requirements could look something like this (TBD):
+
+```yaml
+{
+  "aud": [
+    "vehicle/5GZCZ43D13S812715/kuksa.val.v1"
```

Comment:
see my previous comment. It might not always be desired to have this limited to a simple vehicle. So we should define, 

1. How it looks when the token is valid independent of vehicle (of course signature need to be correct)
  - would we just leave the aud claim out?
  - would it be a special string, i.e. only  "kuksa.val"?
2. regarding "TBD" how it looks like WHEN restricting access, current example given is `vehicle/5GZCZ43D13S812715/kuksa.val.v1`. I think VIN is a good basis (and if a deployment uses that we would expect VIN to live at a well-defined place in our VSS tree? Or would we also add a commandline/runtime configuration for our "vin" indepdendet of loaded VSS -> but yeah, that is a techical quuestion, should not matter for the concept). If choosing VIN, maybe it is good to add some hint that we mean kuksa.val (in case other (SDV) systems also have VIN based tokens), but I think the example is to much: `vehicle`does not add much information in this case, and even specific API as in `kuksa.val.v1` because my assumption is the scopes we have also apply to e.g. the sdv api that is still there, and would also apply to a hypothetical `kuksa.val.v2`, so maybe `kuksa/5GZCZ43D13S812715` is fine?

Reply:
Good points.

I think the document should definitely be updated to clarify this. Perhaps it is even a mistake to include any example of what the `"aud"` claim can contain since it will probably vary between all deployments.

After thinking some more, I think the correct way to view them are as opaque identifiers that should be verified to match what the resource server is configured to accept. The resource server could for example be configured like this:
```
[authorization]
accepted_audiences = ["asd8f98cv789", "234098vcxc"]
```
and it would be up to the infrastructure to configure the resource servers and either issue access tokens with an audience that is accepted by all instances in the fleet, or with unique identifiers for all vehicles in the fleet.

> would we just leave the aud claim out?

The way I understand it, the JWT access token should always include the `"aud"` if we are to follow [RFC9068](https://datatracker.ietf.org/doc/html/rfc9068).

Reply:
"works for me", Implementation-wise, it would imply adding an optiional (commandline?) paameter to databroker, specifying which "aud" it should feel responsible for, right?

Reply:
Yes, exactly.

Reply:
The section still does not say exactly how we are (intending) to implement it, there is unclear wording like  "needs to" "could" and "TBD"

It needs some "KUKSA.val **will** " 

I think we should state
 - if you do not specify otherwise, KUKSA.val will listen to the scope of `kuksa.val` (or whatever else generic term)
 - We can add some wording around "The choice and uniqueness of audiences is a deployment decision. The safest way is using individual audiences for individual Vehicles. Those could be based for example on the VIN. However, depending on the use case, audiences valid for a whole fleet or certain subsets/models of vehicles may be more useful."